examples/fluids/src/misc.c

```diff
@@ -86,23 +86,42 @@ PetscErrorCode DMPlexInsertBoundaryValues_FromICs(DM dm, PetscBool insert_essent
   PetscFunctionReturn(PETSC_SUCCESS);
 }
 
+static PetscErrorCode BinaryReadIntoInt(PetscViewer viewer, PetscInt *out, PetscDataType file_type) {
+  PetscFunctionBeginUser;
+  if (file_type == PETSC_INT32) {
+    PetscInt32 val;
+    PetscCall(PetscViewerBinaryRead(viewer, &val, 1, NULL, PETSC_INT32));
+    *out = val;
+  } else if (file_type == FLUIDS_FILE_TOKEN_64) {
+    PetscInt64 val;
+    PetscCall(PetscViewerBinaryRead(viewer, &val, 1, NULL, PETSC_INT64));
+    *out = val;
+  } else {
+    PetscCall(PetscViewerBinaryRead(viewer, &out, 1, NULL, PETSC_INT));
+  }
+  PetscFunctionReturn(PETSC_SUCCESS);
+}
+
 // @brief Load vector from binary file, possibly with embedded solution time and step number
 PetscErrorCode LoadFluidsBinaryVec(MPI_Comm comm, PetscViewer viewer, Vec Q, PetscReal *time, PetscInt *step_number) {
-  PetscInt   file_step_number;
-  PetscInt32 token;
-  PetscReal  file_time;
+  PetscInt      file_step_number;
+  PetscInt32    token;
+  PetscReal     file_time;
+  PetscDataType file_type = PETSC_INT;
```

Comment:
```suggestion
  PetscDataType file_type = PETSC_INT32;
```

The default needs to be 32bit for binary vectors that were written before the `FLUIDS_FILE_TOKEN` was used. 

I think we _could_ make this more robust by testing the token with 64bit integers like [the PETSc MR does](https://gitlab.com/petsc/petsc/-/merge_requests/7377), but I think that's not necessary; that case is only if we want to read binary vecs written with 64bit ints _and_ weren't written by the fluids code.

Reply:
Oh, I was thinking of `FLUIDS_FILE_TOKEN` before it distinguished 32- vs 64-bit. Were there no old files written with 64-bit indices? I'll defer to your experience with this evolution.

Reply:
At least in the fluids tests, there aren't any built with 64bit ints. I'm also pretty sure that the fluids example was never used with 64bit PETSc before #1234 (mostly because I don't think it was possible before that PR), and fluids already had the `FLUIDS_FILE_TOKEN` by that time.